src/shipit_static_analysis/default.nix

```diff
@@ -0,0 +1,171 @@
+    inProduction = true;
+    version = fileContents ./../../VERSION;
+    src = filterSource ./. { inherit name; };
+    buildInputs =
+      fromRequirementsFile ./requirements-dev.txt python.packages;
+    propagatedBuildInputs =
+      fromRequirementsFile ./requirements.txt python.packages
+      ++ [
+        # Needed for the static analysis
+        clang
+        clang-tools
+        libstdcxx5
+        gcc
+        gcc-unwrapped
+        glibc
+        gawk
+
+        # TODO: Use nixpkgs-mozilla gecko buildInputs
+        # From https://github.com/mozilla/nixpkgs-mozilla/blob/master/pkgs/gecko/default.nix
+        # Expected by "mach"
+        setuptools which autoconf213
+
+        # Expected by the configure script
+        perl unzip zip gnumake yasm pkgconfig
+
+        xlibs.libICE xlibs.libSM xlibs.libX11 xlibs.libXau xlibs.libxcb
+        xlibs.libXdmcp xlibs.libXext xlibs.libXt xlibs.printproto
+        xlibs.renderproto xlibs.xextproto xlibs.xproto xlibs.libXcomposite
+        xlibs.compositeproto xlibs.libXfixes xlibs.fixesproto
+        xlibs.damageproto xlibs.libXdamage xlibs.libXrender xlibs.kbproto
+
+        gnome2.libart_lgpl gnome2.libbonobo gnome2.libbonoboui
+        gnome2.libgnome gnome2.libgnomecanvas gnome2.libgnomeui
+        gnome2.libIDL
+
+        pango
+
+        dbus dbus_glib
+
+        alsaLib libpulseaudio
+        gstreamer gst_plugins_base
+
+        gtk3 glib gobjectIntrospection
+
+        rustStable.rustc rustStable.cargo
+
+        # "mach vendor rust" wants to list modified files by using the vcs.
+        # mercurial is already there !
+        git 
+
+        # needed for compiling cargo-vendor and its dependencies
+        openssl cmake
+      ];
+
+    postInstall = ''
+      mkdir -p $out/tmp
+      mkdir -p $out/bin
+      ln -s ${mercurial'}/bin/hg $out/bin
+      ln -s ${rustStable.rustc}/bin/rustc $out/bin
+      ln -s ${rustStable.cargo}/bin/cargo $out/bin
+      ln -s ${llvmPackages_39.clang-unwrapped}/share/clang/run-clang-tidy.py $out/bin
+			ln -s ${gnumake}/bin/make $out/bin
+			ln -s ${python27}/bin/python2.7 $out/bin
+			ln -s ${gcc}/bin/gcc $out/bin
+			ln -s ${gcc}/bin/g++ $out/bin
+			ln -s ${binutils}/bin/as $out/bin
+			ln -s ${zip}/bin/zip $out/bin
+			ln -s ${perl}/bin/perl $out/bin
+			ln -s ${autoconf213}/bin/autoconf $out/bin/autoconf-2.13
+			ln -s ${gawk}/bin/gawk $out/bin
+			ln -s ${pkgconfig}/bin/pkg-config $out/bin
+    '';
+
+    dockerConfig = { 
+      Env = [
+        "PATH=/bin"
+        "SHELL=/bin/sh"
+        "LANG=en_US.UTF-8"
+        "PKG_CONFIG_PATH=${pango.dev}/lib/pkgconfig:${glib.dev}/lib/pkgconfig:${freetype.dev}/lib/pkgconfig:${fontconfig.dev}/lib/pkgconfig:${cairo.dev}/lib/pkgconfig"
+      ];
+      Cmd = [];
+    };
+
+		shellHook = ''
+			export PATH="${mercurial'}/bin:$PATH"
+			export PATH="${gcc}/bin:$PATH"
+			export PATH="${llvmPackages_39.clang-unwrapped}/share/clang:$PATH"
+
+      export CPATH="${gcc-unwrapped.CPATH}:$CPATH"
```

Comment:
@garbas  In dev. environment run-clang-tidy misses all the standard header files. I tried several things (like exporting a custom `CPATH` or `CPLUS_INCLUDE_PATH`) but this seems strange to have to do that: the nix pkgs for stdcxx5 shoudl expose all this i think ?